backend/kg_generation/prompt_processor.py

```diff
@@ -0,0 +1,132 @@
+from __future__ import annotations
+
+import json
+import re
+
+from langchain_core.prompts import PromptTemplate
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from transformers import PreTrainedTokenizer
+
+jsonRegex = r"\{.*\}"
+
+
+class PromptProcessor:
+    def __init__(
+        self, tokenizer: PreTrainedTokenizer | None = None, num_ctx: int = 8000
+    ) -> None:
+        self.system_prompt = """You are a data scientist working for a company that is building a knowledge graph of Earth and Environment Science.
+        Your task is to extract information from scientific documents and convert it into a knowledge graph. Follow these guidelines:
+            - Provide nodes as: ["entity_id": str, "Type": str, "properties": dict()]
+            - Provide relationships as: ["entity_id_1": str, "hasRelationship": str, "entity_id_2": str, "properties": dict()]
+            - Ensure "entity_id_1" and "entity_id_2" exist as nodes with a matching entity ID.
+            - Do not add relationships with non-existing nodes.
+            - Create a generic "type" for nodes, referring to the http://sweetontology.net/sweetAll ontology for available types and relationships.
+            - Use provided types when possible; create new ones if necessary.
+            - Entity IDs must be lowercase, no special characters or spaces, with words separated by "_".
+            - NO YAPPING before or after your answers. DO NOT add any comment or notes in your answers.
+            - Format your answers to strictly follow the rules in the example below.
+        Example:
+        Data: Temperature measurements taken from the Pacific Ocean show an increase due to global warming.
+        Types: [Process, Property]
+        Nodes: ["temperature_measurements", "Property", {}], ["pacific_ocean", "Realm", {"text": "Pacific Ocean"}], ["increase", "Trend", {"direction": "increase"}], ["global_warming", "Phenomenon", {"text": "Global Warming"}]
+        Relationships: ["temperature_measurements", "hasLocation", "pacific_ocean", {}], ["temperature_measurements", "hasTrend", "increase", {}], ["increase", "isCausedBy", "global_warming", {}]
+        """
+        self.user_prompt_template = PromptTemplate.from_template(
```

Comment:
What does Types in user_prompt_template do? Specifying existing labels?

Reply:
Yes. To reduce the number of entity types.